lib/utils/addEntries.js

```diff
@@ -159,6 +152,79 @@ function addEntries(config, options, server) {
         config.plugins.push(new webpack.HotModuleReplacementPlugin());
       }
     }
+
+    // if there are no additional entries to be added, nothing
+    // below is needed
+    if (!additionalEntries.length) {
+      return;
+    }
+
+    if (
+      webpack.version[0] === '5' &&
+      !compiler.hasWebpackDevServerMakeCallback
+    ) {
+      // this counter allows the dev server to ensure only the most recent
+      // make callback is used, preventing duplicate entry injections
+      if (compiler.webpackDevServerMakeCallbackCount) {
+        // this invalidates old make callbacks
+        compiler.webpackDevServerMakeCallbackCount += 1;
+      } else {
+        compiler.webpackDevServerMakeCallbackCount = 1;
```

Comment:
My idea was to make a counter for callbacks, so that only the most recent callback will be used, fixing the problem I mentioned. Let me know if this seems too hacky.